_health-care/_js/components/HealthCareApp.jsx

```diff
@@ -213,8 +213,30 @@ class HealthCareApp extends React.Component {
     this.setState({ applicationData: newApplicationData });
   }
 
+  resetNullValues(objectData) {
+    const context = this;
+    _.forIn(objectData, (value, key) => {
+      if (value === null) {
+        _.set(objectData, key, '');
+      } else if (typeof (value) === 'object') {
+        context.resetNullValues(value);
+      }
+    });
+
+    return objectData;
+  }
+
   handleContinue() {
-    hashHistory.push(this.getNextUrl());
+    const sectionPath = this.props.location.pathname.split('/').filter((path) => { return !!path; });
+    const sectionData = _.get(this.state.applicationData, sectionPath);
+
+    this.resetNullValues(sectionData);
+
+    this.setState({}, () => {
+      if (validationFunctions.isValidSection(sectionPath, sectionData)) {
+        hashHistory.push(this.getNextUrl());
+      }
```

Comment:
Good question. For now at least I will scroll to the first field with an error, and then perhaps in user testing we can get a better sense of whether that is the right kind of indication, or if we should also include a general error message or notification. Not sure of the most React way of doing this, for now I'll just do it the good old fashioned JavaScript way :)